proposals/release-approval-process.md

```diff
@@ -0,0 +1,26 @@
+
+**Making a release:** OCI specification projects MUST announce intentions to release with two project maintainer sponsors (listed in the repo MAINTAINERS file) on the dev@opencontainers.org mailing list. After the announcement a two-thirds super majority of project maintainers MUST reply to the list with an LGTM within one week for the release to be approved. The maintainers MUST wait a full week for maintainers to reply but if all maintainers reply with an LGTM then the release MAY release earlier except in the case of a major releases.
+
+**Rejecting a release:** A project maintainer MAY choose to reply with REJECT and MUST include a list of concerns filed as GitHub issues. The project maintainers SHOULD try to resolve the concerns and wait for the rejecting maintainer to change their opinion to LGTM. However, a release MAY continue with a single REJECT as long two-thirds of the project maintainers approved the release.
+
+**Timelines:** Specifications have a variety of different timelines in their lifecycle. In early stages the spec SHOULD release often to garner feedback. In later stages there will be bug fix releases, security fix releases, and major breaking change releases. Each of these should have a different level of notification.
+
+- Pre-v1.0.0 specifications SHOULD release on a regular cadence and MUST follow the normal release process. In practice this should be a release every week or two.
+- Major specification releases that introduce new base or optional layers, break backwards compatibility, or add non-optional features MUST release at least two release candidates spaced a minimum of one week apart. In practice this means a major release like a v1.0.0 or v2.0.0 release will take 3 weeks at minimum: one week for v1.0.0-rc1, one week for v1.0.0-rc2, and one week for v1.0.0. Maintainers SHOULD strive to make zero breaking changes during this cycle of release candidates and SHOULD add an additional release candidate when a breaking change is introduced. For example if a breaking change is introduced in -rc2 then a -rc3 SHOULD be made following the normal release process.
+- Minor releases that fix bugs, grammar, introduce optional features, tests, or tooling SHOULD be made on an as-needed basis and MUST follow the normal release process.
+- Security fix releases MUST follow a special release process that substitutes the dev@opencontainers.org email for security@opencontainers.org.
+
+## Conformance/Testing and Applications Releases
```

Comment:
On Thu, Jun 09, 2016 at 10:30:22PM -0700, Aleksa Sarai wrote:
> > +## Conformance/Testing and Applications Releases
>
> I'm confused -- why is this in here at all then? … No rules are
> better than half-baked rules.

I'm all in favor of fully baking the rules.  I'm just trying to
explain why they may not need to be as strict as the spec rules.  I
agree that runC rcs would be nice if cutting a runC release meant
committing to bug-level backwards compat.  But if the compatibility is
covered on the one side by runtime-spec and the other by a
command-line API spec (or similar), then the only remaining consumer
exposure is out-of-spec runC extensions.  If those get the same OCI
weight as in-spec features, then yeah, runC extensions are basically
another spec repository.  If those get less weight than in-spec
features, then you can probably cut a few corners.